runelite-client/src/main/java/net/runelite/client/plugins/statusorbs/StatusOrbsOverlay.java

```diff
@@ -222,4 +265,61 @@ private void renderRegen(Graphics2D g, WidgetInfo widgetInfo, double percent, Co
 		g.setColor(color);
 		g.draw(arc);
 	}
+
+	public void updateSpec()
```

Comment:
Wasn't there something showing a circle around the orb already? Can't you reuse that?

Reply:
this sets a tool tip with available spec and time until you regen for next spec. its similar but different to what currently exists

Reply:
I get that, but that doesn't mean you should calculate the time to next spec refill twice :P

(more comments coming)